packages/calypso-e2e/src/lib/pages/site-editor-page.ts

```diff
@@ -0,0 +1,170 @@
+	editorIframe: `iframe.is-loaded[src*="${ wpAdminPath }"]`,
+	editorRoot: 'body.block-editor-page',
+	editorCanvasIframe: 'iframe[name="editor-canvas"]',
+	editorCanvasRoot: '.wp-site-blocks',
+	templateLoadingSpinner: '[aria-label="Block: Template Part"] .components-spinner',
+};
+
+/**
+ * Represents an instance of the FSE site editor.
+ * This class is composed of editor components, combining them into larger flows.
+ */
+export class SiteEditorPage {
+	private page: Page;
+
+	private editorToolbarComponent: EditorToolbarComponent;
+	private editorGutenbergComponent: EditorGutenbergComponent;
+	private editorSidebarBlockInserterComponent: EditorSidebarBlockInserterComponent;
+	private editorWelcomeTourComponent: EditorWelcomeTourComponent;
+
+	/**
+	 * Constructs an instance of the page POM class.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+
+		this.editorToolbarComponent = new EditorToolbarComponent( page, this.editor );
+		this.editorWelcomeTourComponent = new EditorWelcomeTourComponent( page, this.editor );
+		this.editorSidebarBlockInserterComponent = new EditorSidebarBlockInserterComponent(
+			page,
+			this.editor
+		);
+		// Because of the unique extra iframe in the site editor, this component needs the canvas locator.
+		this.editorGutenbergComponent = new EditorGutenbergComponent( page, this.editorCanvas );
+	}
+
+	/**
+	 * A frame-safe locator to the top level element in the editor, for building other editor locators.
+	 */
+	private get editor(): Locator {
+		return this.page.url().includes( wpAdminPath )
+			? this.page.locator( selectors.editorRoot )
+			: this.page.frameLocator( selectors.editorIframe ).locator( selectors.editorRoot );
+	}
+
+	/**
+	 * A frame-safe locator to the top level element in the editor canvas, for building other editor locators.
+	 * The editor canvas is an iframe wrapping the center editor area that is unique to the site-editor.
+	 */
+	private get editorCanvas(): Locator {
+		return this.editor
+			.frameLocator( selectors.editorCanvasIframe )
+			.locator( selectors.editorCanvasRoot );
+	}
+
+	/**
+	 * Visit the site editor by URL directly.
+	 *
+	 * @param siteHostName Host name of the site, without scheme. (e.g. testsite.wordpress.com)
+	 */
+	async visit( siteHostName: string ): Promise< void > {
+		await this.page.goto( getCalypsoURL( `site-editor/${ siteHostName }` ) );
+	}
+
+	/**
+	 * Waits until the site editor is fully loaded.
+	 */
+	async waitUntilLoaded(): Promise< void > {
+		// There are more stages to the site editor loading than the regular editor.
+		// The most reliable "last" thing to load is the canvas iframe
+		await this.editorCanvas.waitFor( { timeout: 60 * 1000 } );
+		// But then, template parts load async afterwards!
+		const spinnerLocator = this.editorCanvas.locator( selectors.templateLoadingSpinner );
+		// There could be many spinners, so we will keep waiting for the first to be detached.
```

Comment:
Sounds good! I agree. Personally, I find it frustrating that you can't turn off strict mode for locators. 😞  This has been stable so far (I think because it all completes in the same async operation), and it's nice to use the same API throughout, but I will watch it and revert if needed. 👍